README.md

```diff
@@ -412,6 +418,53 @@ When collecting, only the lexically last flattened array will get any data, and
 
 In this case, the `$other` property has two keys, `foo` and `bar`, with values `beep` and `boop`, respectively.  The same JSON will deserialize back to the same object as before.
 
+With flattened object the serialization and deserialization is asymmetrical, consider this example:
+```php
+class Page
+{
+    public function __construct(
+        public int $total,
+        public int $limit,
+        #[Serde\Field(flatten: true)]
+        public null|PaginationState $state,
+    ) {}
+}
+
+class PaginationState
+{
+    public function __construct(
+        public int $offset,
+    ) {
+    }
+}
+$page = new Page(100, 10, null);
+```
+
+Will serialized with omitted the flattened object into this JSON payload
+
+```json
+{
+    "total":100,
+    "limit":10
+}
+```
+
+However, this payload will be deserialized intot the following object, especially notice the uninitialized DTO:
```

Comment:
```suggestion
However, when deserializing back into PHP, the `$state` property will get a new, uninitialized `PaginationState` instance created.  If you have a nullable property, be aware that it may end up with an uninitialized object.
```

Reply:
That said... I really don't like this behavior.  It seems like a foot gun waiting to happen.